ps/ps.go

```diff
@@ -445,9 +451,16 @@ func JoinNamespaceAndProcessInfo(pid, format string) ([]string, error) {
 		return nil, err
 	}
 
-	hostProcesses, err = getHostProcesses(pid)
-	if err != nil {
-		return nil, err
+	// extract data from host processes only on-demand / when at least one
+	// of the specified descriptors requires host data
+	for _, d := range formatDescriptors {
+		if d.onHost {
+			hostProcesses, err = getHostProcesses(pid)
```

Comment:
Could you check if hostProcesses already set before calling this multiple times.

podman pod cid hgroup hpid huser

Will read from the system three times, I believe.

Reply:
We avoid that with the `break` in line 462, so we read at most once.